src/test/java/unit/com/mathworks/ci/actions/RunMatlabCommandActionTest.java

```diff
@@ -38,17 +55,55 @@ public class RunMatlabCommandActionTest {
     public void init() {
         if (!setup) {
             setup = true;
-            action = new RunMatlabCommandAction(runner, params);
+            action = new RunMatlabCommandAction(runner, annotator, params);
+
+            when(runner.getTempFolder()).thenReturn(tempFolder);
+            when(tempFolder.getRemote()).thenReturn("/path/less/traveled");
+
+            when(params.getTaskListener()).thenReturn(listener);
+            when(listener.getLogger()).thenReturn(out);
+
+            when(params.getBuild()).thenReturn(build);
         }
     }
 
+    @Test
+    public void shouldCopyPluginsToTempDirectory() throws IOException, InterruptedException, MatlabExecutionException {
+        action.run();
+
+        InOrder inOrder = inOrder(runner);
+
+        inOrder.verify(runner)
+            .copyFileToTempFolder(MatlabBuilderConstants.DEFAULT_PLUGIN, MatlabBuilderConstants.DEFAULT_PLUGIN);
```

Comment:
Same as above combining these might help us re writing it again.

Reply:
Moved the test to `RunMatlabActionTest.java`